controllers/cindervolume_controller.go

```diff
@@ -580,14 +580,31 @@ func (r *CinderVolumeReconciler) generateServiceConfigs(
 	}
 	customData[cinder.CustomServiceConfigSecretsFileName] = customSecrets
 
+	templateParameters := make(map[string]interface{})
+
+	networkAttachmentAddrs := cinder.GetNetworkAttachmentAddrs(
+		instance.Namespace, instance.Spec.NetworkAttachments, instance.Status.NetworkAttachments)
+
+	// Configure the target IP addresses. These are only used by the LVM backend,
+	// but it's safe (and easier) to always configure them regardless of the
+	// backend type. The configuration also relies on the fact that the LVM
+	// backend must have only one replica.
+	if len(networkAttachmentAddrs) > 0 {
```

Comment:
Would you consider doing this conditionally only when using LVM backends?

The idea would be to change the `extendCustomServiceConfig` that is being called in  L556 to return 2 values, the `customServiceConfig` and a boolean for `usesLVM`. We may need to change the method name.

The `extendCustomServiceConfig` shouldn't need a lot of code changes, basically:
- Not `break` on `token == "enabled_backends"`
- Create a new variable `hasLVM := false` and another `foundDrivers := 0`
- Get the `value` as well as the `token` we are getting on `SplitN`
- Add a new `} else if token == "volume_driver" && value == "cinder.volume.drivers.lvm.LVMVolumeDriver" {` where we set `hasLVM` to `true` and increase the `foundDrivers` counter
- At the end the actual boolean that determines whether there's LVM or not would be `hasLVM && len(backendNames) == `foundDrivers` because we have to account that it's the default driver and its value could be implicit.

The reason for this is to avoid restarting the container when not using LVM, which are all the production environments.

Reply:
Good idea, done!